#! /bin/sh -e

# DP: Workaround for Scratchbox issue regarding foreign toolchains not working
# DP: in forked processes, which is breaking compilation of some extensions.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Lib/distutils/command/build_ext.py	2008-10-30 16:46:21.000000000 -0300
+++ Lib/distutils/command/build_ext.py	2008-10-30 16:48:39.000000000 -0300
@@ -412,6 +412,9 @@
         # First, sanity-check the 'extensions' list
         self.check_extensions_list(self.extensions)
 
+        # Hack to get compilation working in newer maemo targets
+        os.environ['LD_PRELOAD'] = os.environ.get('SBOX_PRELOAD')
+
         for ext in self.extensions:
             self.build_extension(ext)
 
